server/app/services/applicant/ApplicantService.java

```diff
@@ -1216,9 +1221,28 @@ private ApplicationPrograms relevantProgramsForApplicantInternal(
         .build();
   }
 
+  private ProgramDefinition getProgramDefinitionForDraftApplication(
+      ImmutableList<ProgramDefinition> programList, long programId, Request request) {
+
+    if (settingsManifest.getFastforwardEnabled(request)) {
+      logger.debug("Fast-forwarding is enabled, but we aren't doing anything special here yet.");
+    }
+
+    // Get the program definition from the all programs list, since that has the
+    // associated question data.
+    return findProgramWithId(programList, programId);
+  }
+
   private ProgramDefinition findProgramWithId(
       ImmutableList<ProgramDefinition> programList, long id) {
-    return programList.stream().filter(p -> p.id() == id).findFirst().get();
+    return programList.stream()
+        .filter(p -> p.id() == id)
+        .findFirst()
+        .orElseThrow(
+            () ->
+                new NoSuchElementException(
+                    String.format(
+                        "Expected to find program id %s in this list, but it was not found.", id)));
```

Comment:
I got sick of this throwing "no such element exception, dunno what to tell you :woman_shrugging: ", so I'm rethrowing with the id that was used.